target/linux/ramips/image/mt7621.mk

```diff
@@ -684,10 +684,12 @@ define Device/dlink_dir-8xx-a1
   DEVICE_VENDOR := D-Link
   DEVICE_PACKAGES := kmod-mt7615-firmware -uboot-envtools
   KERNEL := $$(KERNEL) | uimage-sgehdr
-  IMAGES += factory.bin
+  IMAGES += recovery.bin factory.bin
   IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | \
 	check-size | append-metadata
-  IMAGE/factory.bin := append-kernel | append-rootfs | check-size
+  IMAGE/recovery.bin := append-kernel | append-rootfs | check-size
+  IMAGE/factory.bin := append-kernel | append-rootfs | check-size | \
```

Comment:
Why not start from IMAGE/recovery.bin?

Reply:
makes sense, at least for devices with this image format it could be generated incrementally

Reply:
so change it to 
IMAGE/factory.bin := $$(IMAGE/recovery.bin) | dlink-sge-image $$$$(DEVICE_MODEL)
?

Reply:
I think this should work.

Reply:
BTW, since we have the following in `target/linux/ramips/image/Makefile`:
```
define Device/Default
  [...]
  IMAGES := sysupgrade.bin
  COMPILE :=
  sysupgrade_bin := append-kernel | append-rootfs | pad-rootfs
  IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | check-size | append-metadata
endef
```
You could drop the `IMAGE/sysupgrade.bin` line.

Reply:
ok Dropped